### PR TITLE
MAISTRA-2425 move MeshFederation CertificateChain to ConfigMap refere…

### PR DESCRIPTION
**Pull request by rcernich**
…nce; add ability to specify ports for remote service and discovery

Signed-off-by: rcernich <rcernich@redhat.com>

### CODE REVIEW

File: core/v1/meshfederation_types.go [hunk 1]
```diff
@@ -80,14 +80,16 @@ type MeshFederationSecurity struct {
...
-	// Name of secret containing certificate chain to be used to validate
-	// the remote.  This is also used to validate certificates used by the
-	// remote services (both client and server certificates).
-	// XXX: maybe this is only used to initiate a connection, with the actual
-	// certs stored in the status field, as retrieved from the remote mesh, or
-	// maybe this identifies an endpoint used to retrieve a cert chain, a la jwks
+	// Name of ConfigMap containing certificate chain to be used to
+	// validate the remote.  This is also used to validate certificates used by
+	// the remote services (both client and server certificates).  The name of
+	// the entry should be of the form: <federation-name>-ca.crt, e.g. if the
```
Comment (dgn) [inline core/v1/meshfederation_types.go]: why the name requirement? Are we planning to support storing multiple root certs in the same ConfigMap? If not, I think just using a generic name like `root-cert.pem` is fine

File: core/v1/meshfederation_types.go [hunk 2]
```diff
@@ -80,14 +80,16 @@ type MeshFederationSecurity struct {
...
+	// MeshFederation resource is named remote-mesh, the entry in the ConfigMap
+	// should be named remote-mesh-ca.crt.
+	// XXX: should we use the mesh federation name, or should we use the
+	// trustDomain as the key?
 	// +optional
-	CertificateChain string `json:"certificateChain,omitempty"`
+	CertificateChain corev1.TypedLocalObjectReference `json:"certificateChain,omitempty"`
```
Comment (dgn) [inline core/v1/meshfederation_types.go:90]: nit: should we rename the field to make it more obvious that it's referencing a ConfigMap?

**Review by dgn — COMMENTED** [review]
lgtm, just a few questions

File: core/v1/meshfederation_types.go (hunk 1 above)
Comment (rcernich) [reply core/v1/meshfederation_types.go]: Good question.  I don't have a strong opinion one way or the other.  I didn't give a lot of thought to the user workflow, other than thinking maybe having a single resource with the CAs for all federated clusters would be nice, as opposed to individual resources for each CA.  Having a generic name seems simpler for the user (no need to figure out what name to use).  Should I make the entry key configurable as well?

File: core/v1/meshfederation_types.go (hunk 2 above)
Comment (rcernich) [reply core/v1/meshfederation_types.go:90]: I left this in place in case we wanted to support secrets in the future.

**Review by brian-avery — APPROVED** [review]

**Pull request merged**